impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -192,9 +226,12 @@ std::optional<Snapshot> DirectionalGaussianBlurFilterContents::RenderFilter(
   sampler_desc.min_filter = MinMagFilter::kLinear;
   sampler_desc.mag_filter = MinMagFilter::kLinear;
 
-  return Snapshot{.texture = out_texture,
-                  .transform = Matrix::MakeTranslation(coverage.origin),
-                  .sampler_descriptor = sampler_desc};
+  return Snapshot{
+      .texture = out_texture,
+      .transform = texture_rotate.Invert() *
+                   Matrix::MakeTranslation(pass_texture_rect.origin) *
+                   Matrix::MakeScale(Vector2(1 / x_scale, 1)),
+      .sampler_descriptor = sampler_desc};
```

Comment:
This seems pretty testable to me. If it's currently difficult to unit test a RenderFilter I think we should land this without a test, but we should be able to write some tests against this...

Reply:
Seems reasonable... is it possible to test private interfaces with gtest? I would prefer to keep these internals out of the public API.

Reply:
Added an issue to make this stuff testable https://github.com/flutter/flutter/issues/109795

Reply:
If you make the test fixture a friend you can. If this isn't the right spot to test maybe there's a public method somewhere in the chain we could test?

Reply:
> If this isn't the right spot to test maybe there's a public method somewhere in the chain we could test?

I think it makes sense to find a way to make this specific barrier testable (albeit this API will undergo more rewrites).

From an Entities API behavior testing perspective, the clear quantifiable/stable/portable barrier between the user's inputs and colors on the screen is the set of command buffers that Contents populate when they render. Once we have a good story for mocking the render layer, that'll probably be enough to cover most contents reasonably well. But filters are more complicated and we should definitely test behavior of the internal API to detect subtle breakages that will be hard to nail down through a command buffer predicate test.